adapters/repos/db/shard.go

```diff
@@ -355,6 +357,46 @@ func (s *Shard) ObjectCountAsync() int {
 	return b.CountAsync()
 }
 
+// ForEachVectorIndex iterates through each vector index configured in the shard (named and legacy).
+// Iteration stops at the first return of non-nil error.
+func (s *Shard) ForEachVectorIndex(f func(targetVector string, index VectorIndex) error) error {
+	for name, idx := range s.vectorIndexes {
```

Comment:
here also let's use `targetVector` instead of `name`, let's make this change also in `ForEachVectorQueue` method